fmn/rules/tracking_rules.py

```diff
@@ -36,35 +36,23 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.usernames = set(self._params)
 
-    def matches(self, message):
-        for package in message.packages:
-            if self.usernames & set(self._requester.get_package_owners(package)):
-                return True
-        for container in message.containers:
-            if self.usernames & set(self._requester.get_container_owners(container)):
-                return True
-        for module in message.modules:
-            if self.usernames & set(self._requester.get_module_owners(module)):
-                return True
-        for flatpak in message.flatpaks:
-            if self.usernames & set(self._requester.get_flatpak_owners(flatpak)):
-                return True
+    async def matches(self, message):
+        for artifact_type in ArtifactType:
+            for artifact in getattr(message, artifact_type.name):
+                owners = await self._requester.distgit.get_owners(
+                    artifact_type.value, artifact, "user"
+                )
+                if self.usernames & set(owners):
+                    return True
         return False
 
-    def prime_cache(self, cache):
+    async def prime_cache(self, cache):
         for username in self.usernames:
-            cache["packages"].update(
-                self._requester.get_owned_by_user(artifact_type="packages", name=username)
-            )
-            cache["containers"].update(
-                self._requester.get_owned_by_user(artifact_type="containers", name=username)
-            )
-            cache["modules"].update(
-                self._requester.get_owned_by_user(artifact_type="modules", name=username)
-            )
-            cache["flatpaks"].update(
-                self._requester.get_owned_by_user(artifact_type="flatpaks", name=username)
-            )
+            for artifact_type in ArtifactType:
+                owned = await self._requester.distgit.get_owned(
+                    artifact_type.value, username, "user"
+                )
+                getattr(cache, artifact_type.name).update(set(owned))
```

Comment:
This looks almost if it would profit from `PagureAsyncProxy.get_projects()` in my PR – it would let us request all artifacts of a user in one go and we could sort them out locally.